### PR TITLE
Basic auth fixes

### PR DESCRIPTION
**Pull request by mohitdeuex**
### Describe your changes :

Fixed Message for Login #7701
 #7702

image

image
image

image

image

#
### Type of change :

- [x] Improvement
- [x] New feature

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/Users/ChangePasswordForm.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+}) => {
+  const [form] = Form.useForm();
+  const newPassword = Form.useWatch('newPassword', form);
+
+  const validationMessages = useMemo(
+    () => ({
+      required: '${label} is required',
+    }),
+    []
+  );
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/Users/ChangePasswordForm.tsx:25]: You can use it from: https://github.com/open-metadata/OpenMetadata/blob/92aea1c1798679a1318950d2233840d74296dfea/openmetadata-ui/src/main/resources/ui/src/constants/auth.constants.ts#L16

File: openmetadata-ui/src/main/resources/ui/src/components/Users/ChangePasswordForm.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            {
+              validator: (_, value) => {
+                if (isEmpty(newPassword)) {
+                  return Promise.reject('Please type New Password first');
+                }
+                if (value !== newPassword) {
+                  return Promise.reject("Password doesn't match");
+                }
+
+                return Promise.resolve();
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/Users/ChangePasswordForm.tsx:86]:
```suggestion
               if (value !== newPassword) {
                  return Promise.reject("Password doesn't match");
                }

                return Promise.resolve();
```

File: openmetadata-ui/src/main/resources/ui/src/components/Users/Users.component.tsx [hunk 3]
```diff
@@ -299,7 +326,13 @@ const Users = ({
...
     if (isAdminUser || isLoggedinUser || isAuthDisabled) {
       return (
-        <div className="tw--ml-5 tw-flex tw-items-center tw-justify-between tw-px-3">
+        <div
+          className={classNames(
+            'tw--ml-5 tw-flex tw-items-center tw-justify-between tw-px-3',
+            {
+              'm-b-md': !isAuthProviderBasic || !isLoggedinUser,
+            }
+          )}>
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/Users/Users.component.tsx:335]: Other components should not get affected. You can add spacing to Change password component itself

File: openmetadata-ui/src/main/resources/ui/src/components/Users/Users.component.tsx [hunk 4]
```diff
@@ -324,6 +357,24 @@ const Users = ({
     }
   };
 
+  const getChangePasswordComponent = () => {
+    return (
+      <div className="m-b-sm tw-px-3">
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/Users/Users.component.tsx:362]: Add spacing here instead

File: openmetadata-ui/src/main/resources/ui/src/components/Users/Users.component.tsx [hunk 5]
```diff
@@ -324,6 +357,24 @@ const Users = ({
...
+        <Typography.Text
+          className="tw-text-primary tw-text-xs tw-cursor-pointer"
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/Users/Users.component.tsx:364]: Can we avoid tailwind classes?

**Comment by chirag-madlani** [comment]
@Ashish8689 Can you attach some screenshot for Change password UI changes?

**Comment by Ashish8689** [comment]
@chirag-madlani  Updated files as per comments!

File: openmetadata-ui/src/main/resources/ui/src/components/Users/Users.component.tsx [hunk 6]
```diff
@@ -225,14 +237,28 @@ const Users = ({
     }
   };
 
+  const handleChangePassword = async (data: ChangePasswordRequest) => {
+    try {
+      await changePassword(data);
+      showSuccessToast(
```
[the remark this answers is not in the document]
Comment (Ashish8689) [reply openmetadata-ui/src/main/resources/ui/src/components/Users/Users.component.tsx:244]: Updated!

File: openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/UserRepository.java [hunk 7]
```diff
@@ -47,8 +47,8 @@
 
 @Slf4j
 public class UserRepository extends EntityRepository<User> {
-  static final String USER_PATCH_FIELDS = "profile,roles,teams,authenticationMechanism,isEmailVerified";
-  static final String USER_UPDATE_FIELDS = "profile,roles,teams,authenticationMechanism,isEmailVerified";
+  static final String USER_PATCH_FIELDS = "profile,roles,teams,isEmailVerified";
```
Comment (mohitdeuex) [inline openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/UserRepository.java:50]: @nahuelverdugo  will this impact the BOTS flow??

File: openmetadata-ui/src/main/resources/ui/src/components/CreateUser/CreateUser.component.tsx [hunk 8]
```diff
@@ -230,10 +273,28 @@ const CreateUser = ({
...
 
+  //   *******  Generate Random Passwprd  *****
+  const generateRandomPassword = async () => {
+    setIsPasswordGenerating(true);
+    try {
+      const password = await generateRandomPwd();
+      setTimeout(() => {
+        setGeneratedPassword(password);
+        form.setFieldsValue({ generatedPassword: password });
+      }, 500);
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/CreateUser/CreateUser.component.tsx:284]: Why settimeout?

**Review by chirag-madlani — APPROVED** [review]

**Review by nahuelverdugo — APPROVED** [review]

**Pull request merged**